### PR TITLE
Make ElasticsearchException ids for 2.0 consistent with master

### PR DESCRIPTION
**Pull request by jasontedor**
This commit reworks the ElasticsearchException ids so that they are
consistent with master at f40ae25352f842f188ce833406f8aeb500ae64ca.

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/ExceptionSerializationTests.java [hunk 1]
```diff
@@ -623,4 +623,170 @@ public UnknownException(String message, Throwable cause) {
...
+        ids.put(129, org.elasticsearch.index.engine.EngineClosedException.class);
+        ids.put(130, org.elasticsearch.action.NoShardAvailableActionException.class);
+        ids.put(131, org.elasticsearch.action.UnavailableShardsException.class);
+        ids.put(132, org.elasticsearch.index.engine.FlushFailedEngineException.class);
+        ids.put(133, org.elasticsearch.common.breaker.CircuitBreakingException.class);
+        ids.put(134, org.elasticsearch.transport.NodeNotConnectedException.class);
+        ids.put(135, org.elasticsearch.index.mapper.StrictDynamicMappingException.class);
+        ids.put(136, org.elasticsearch.action.support.replication.TransportReplicationAction.RetryOnReplicaException.class);
+        ids.put(137, org.elasticsearch.indices.TypeMissingException.class);
+        ids.put(138, null);
```
Comment (s1monw) [inline core/src/test/java/org/elasticsearch/ExceptionSerializationTests.java]: can't we make those be filled by 2.0 and then fix master accordingly?

**Comment by s1monw** [comment]
left one comment

File: core/src/test/java/org/elasticsearch/ExceptionSerializationTests.java (hunk 1 above)
Comment (jasontedor) [inline core/src/test/java/org/elasticsearch/ExceptionSerializationTests.java]: Done.

**Comment by s1monw** [comment]
LGTM

**Pull request merged**